From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Mon, 14 Mar 2016 19:15:48 +0100
Subject: [PATCH] Disable relax relocations for x86 loader
Patch-mainline: Never, a temporary workaround for 32bit builds
References: bsc#970239

The 32-bit loader for decompression cannot relocate itself
if relax relocations (R_386_GOT32X) are in use,
as LD optimizes them in an incompatible way.
This patch works around this issue by disabling those relocations.
However, this will only work until the next binutils release
optimizes non-relax relocations in the same way.

Signed-off-by: Fabian Vogt <fabian@ritter-vogt.de>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 arch/x86/boot/compressed/Makefile |    1 +
 1 file changed, 1 insertion(+)

--- a/arch/x86/boot/compressed/Makefile
+++ b/arch/x86/boot/compressed/Makefile
@@ -30,6 +30,7 @@ KBUILD_CFLAGS += $(cflags-y)
 KBUILD_CFLAGS += -mno-mmx -mno-sse
 KBUILD_CFLAGS += $(call cc-option,-ffreestanding)
 KBUILD_CFLAGS += $(call cc-option,-fno-stack-protector)
+KBUILD_CFLAGS += $(call as-option,-Wa$(comma)-mrelax-relocations=no,)
 
 KBUILD_AFLAGS  := $(KBUILD_CFLAGS) -D__ASSEMBLY__
 GCOV_PROFILE := n
